<?xml version="1.0"?>
<launch>
    <group ns="aura">
		<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
		 <!--<rosparam file="$(find setup)/config/costmap_common_params.yaml" command="load" ns="global_costmap"/>-->
	     <!--<rosparam file="$(find setup)/config/costmap_common_params.yaml" command="load" ns="local_costmap"/>-->
	     <!--rosparam file="$(find setup)/config/local_costmap_params.yaml" command="load" /-->
	     <!--rosparam file="$(find setup)/config/global_costmap_params.yaml" command="load" /-->
	     <!--rosparam file="$(find setup)/config/base_local_planner_params.yaml" command="load" /-->
	  <remap from="cmd_vel" to="/cmd_vel_mux/input/navi"/>
      <remap from="map" to="base_map" />
      <remap from="odom" to="odom" />
      <param name="base_global_planner" type="string" value="global_planner/GlobalPlanner" />
      <param name="base_local_planner" type="string" value="base_local_planner/TrajectoryPlannerROS" />
      <param name="conservative_reset_dist" type="double" value="3.0" />
      <param name="controller_frequency" type="double" value="7.5" />
      <param name="planner_frequency" type="double" value="1"/>
      <rosparam>
        local_costmap:
          #footprint: [[0.28, 0.28], [-0.28, 0.28], [-0.28, -0.28], [0.28, -0.28]]
          robot_radius : 0.16
          global_frame: aura/odom
          robot_base_frame: aura/base_link
          static_map: false
          rolling_window: true
          width: 2
          height: 2
          resolution: 0.05


          plugins:
             - {name: static,                    type: "costmap_2d::StaticLayer"}
             - {name: inflation,                 type: "costmap_2d::InflationLayer"}
             - {name: obstacle,                 type: "costmap_2d::ObstacleLayer"}
      </rosparam>
      <rosparam>
        global_costmap:
          #footprint: [[0.28, 0.28], [-0.28, 0.28], [-0.28, -0.28], [0.28, -0.28]]
          robot_radius : 0.16
          global_frame: aura/map
          robot_base_frame: aura/base_link
          resolution: 0.2
          static_map: true
          rolling_window: false
          width: 15.0
          height: 15.0

          plugins:
             - {name: static,                    type: "costmap_2d::StaticLayer"}
             - {name: inflation,                 type: "costmap_2d::InflationLayer"}
             - {name: obstacle,                  type: "costmap_2d::ObstacleLayer"}
      </rosparam>
 <rosparam>
         TrajectoryPlannerROS:
           #Robot Configuration Parameters
             max_vel_x: 2 #0.4
             min_vel_x: 0.02
             max_vel_theta: 2.3
             min_vel_theta: -2.3
             min_in_place_vel_theta: 0.4
             escape_vel: -0.15
           #Forward Simulation Parameters
             sim_time: 1.2
             vx_samples: 8
             vtheta_samples: 26
             sim_granularity: 0.01
             angular_sim_granularity: 0.015

             acc_lim_theta: 1
             acc_lim_x: 0.5 #0.3
             acc_lim_y: 0.0

             holonomic_robot: false

           #Goal Tolerance Parameters
             yaw_goal_tolerance: 0.05 #default 0.05
             xy_goal_tolerance: 0.5 #default 0.1
           #Trajectory Scoring Parameters
             pdist_scale:  1
             gdist_scale:  0.8
             occdist_scal: 0.01 #default 0.01 #pre 0.075

         NavfnROS:
           allow_unknown: true
           default_tolerance: 0.4
           visualize_potential: true   # false

         GlobalPlanner:
             allow_unknown: true
             use_dijkstra: true
             use_quadratic: false
             visualize_potential: true
             use_grid_path: false
             old_navfn_behavior: false
             cost_factor: 3
             neutral_cost: 50
             lethal_cost: 253
             default_tolerance: 0
             
         TebLocalPlannerROS:

            odom_topic: fake_odom
            map_frame: /map

            # Trajectory

            teb_autosize: true
            dt_ref: 0.3
            dt_hysteresis: 0.1
            global_plan_overwrite_orientation: true
            max_global_plan_lookahead_dist: 5.0
            feasibility_check_no_poses: 5

            # Robot

            max_vel_x: 0.6
            max_vel_x_backwards: 0.15
            max_vel_theta: 1.2
            acc_lim_x: 0.8
            acc_lim_theta: 1
            min_turning_radius: 0.115
            inflation_dist: 40
            footprint_model: # types: "point", "circular", "two_circles", "line", "polygon"
                type: "polygon"
                radius: 0.16 # for type "circular"
                #line_start: [-0.3, 0.0] # for type "line"
                #line_end: [0.3, 0.0] # for type "line"
                #front_offset: 0.2 # for type "two_circles"
                #front_radius: 0.2 # for type "two_circles"
                #rear_offset: 0.2 # for type "two_circles"
                #rear_radius: 0.2 # for type "two_circles"
                #vertices: [ [0.3602, -0.0708], [0.274, -0.0708], [0.274, -0.254], [-0.274, -0.274], [-0.274, 0.274], [0.274, 0.274], [0.274, 0.0708], [0.3602, 0.0708] ] # for type "polygon"

            # GoalTolerance

            xy_goal_tolerance: 0.1
            yaw_goal_tolerance: 0.5
            free_goal_vel: false

            # Obstacles

            min_obstacle_dist: 0.18
            include_costmap_obstacles: true
            costmap_obstacles_behind_robot_dist: 1.5
            obstacle_poses_affected: 30
            costmap_converter_plugin: ""
            costmap_converter_spin_thread: true
            costmap_converter_rate: 5

            # Optimization

            no_inner_iterations: 5
            no_outer_iterations: 4
            optimization_activate: true
            optimization_verbose: false
            penalty_epsilon: 0.3
            weight_max_vel_x: 1
            weight_max_vel_theta: 1
            weight_acc_lim_x: 0.5
            weight_acc_lim_theta: 0.5
            weight_kinematics_nh: 1000
            weight_kinematics_forward_drive: 1
            weight_kinematics_turning_radius: 1
            weight_optimaltime: 1
            weight_obstacle: 50
            weight_dynamic_obstacle: 10 # not in use yet
            alternative_time_cost: false # not in use yet

            # Homotopy Class Planner

            enable_homotopy_class_planning: true
            enable_multithreading: true
            simple_exploration: false
            max_number_classes: 4
            roadmap_graph_no_samples: 15
            roadmap_graph_area_width: 5
            h_signature_prescaler: 0.5
            h_signature_threshold: 0.1
            obstacle_keypoint_offset: 0.1
            obstacle_heading_threshold: 0.45
            visualize_hc_graph: false


       </rosparam>
		</node>
		</group>
</launch>
